modules/op-signing-secrets-in-tekton-chains.adoc

```diff
@@ -0,0 +1,60 @@
+// This module is included in the following assembly:
+//
+// *cicd/pipelines/using-tekton-chains-for-pipelines-supply-chain-security.adoc
+
+:_content-type: CONCEPT
+[id="signing-secrets-in-tekton-chains_{context}"]
+= Signing secrets in Tekton Chains
+
+Cluster administrators can generate a key pair and use Chains to sign artifacts using a Kubernetes secret. For Chains to work, a private key and a password for encrypted keys must exist as part of the `signing-secrets` Kubernetes secret, in the `tekton-chains` namespace.
+
+Currently, Chains supports the `x509` and `cosign` signature schemes.
+
+[NOTE]
+====
+Use only one of the supported signature schemes.
+====
+
+[id="chains-signing-secrets-x509_{context}"]
+== Signing using x509
+
+To use the `x509` signing scheme with Chains, store the `x509.pem` private key of the `ed25519` or `ecdsa` type in the `signing-secrets` Kubernetes secret. Ensure that the key is stored as an unencrypted PKCS8 PEM file (`BEGIN PRIVATE KEY`).
+
+[id="chains-signing-secrets-cosign_{context}"]
+== Signing using cosign
+
+To use the `cosign` signing scheme with Chains:
+
+. link:https://docs.sigstore.dev/cosign/installation/[Install] cosign.
+
+. Generate the `cosign.key` and `cosign.pub` key pairs.
++
+[source,terminal]
+----
+$ cosign generate-key-pair k8s://tekton-chains/signing-secrets
+----
++
+Cosign prompts you for a password, and creates a Kubernetes secret.
+
+. Store the encrypted `cosign.key` private key and the `cosign.password` decryption password in the `signing-secrets` Kubernetes secret. Ensure that the private key is stored as an encrypted PEM file of the `ENCRYPTED COSIGN PRIVATE KEY` type.
+
+[id="chains-troubleshooting-signing_{context}"]
+== Troubleshooting signing
+
+If the signing secrets are already populated, you may get the following error:
```

Comment:
IBM SG: use `might` instead of `may`

Reply:
As I have said against other comments - creating separate procedures for such single-step stuff means I will have to refactor them into separate modules. Honestly, at this point, I would rather not do that. Any way, I am going to do a massive refactoring of the entire Pipelines docs after the GA, so I will do these things then.